_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,169 @@
+
+Your index's structure can help you decide which normalization technique to use, as each has its own advantages. If your documents are highly similar and a typical query returns top-k results that are closely clustered together within the index (as shown in the following graph), min-max normalization may be the better choice.
+
+![Image for min-max distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost1.jpg){: .img-fluid}
+
+However, z-score normalization is a better choice if the results are more evenly distributed and follow a pattern similar to a normal distribution, as shown in the following graph.
+
+![Image for zscore distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost2.jpg){: .img-fluid}
+
+The basic process for choosing between min-max and z-score normalization is outlined in the following diagram.
+
+![Image for flow](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost3.png){: .img-fluid}
+
+## How to use z-score normalization
+
+To use z-score normalization, create a search pipeline and specify `z_score` as the `technique`:
+
+```json
+PUT /_search/pipeline/z_score-pipeline
+{
+    "description": "Zscore processor for hybrid search",
+    "phase_results_processors": [
+        {
+            "normalization-processor": {
+                "normalization": {
+                    "technique": "z_score"
+                }
+            }
+        }
+    ]
+}
+```
+
+Next, create a hybrid query and apply the pipeline to the query:
+
+```json
+POST my_index/_search?search_pipeline=z_score-pipeline
+{
+   "query": {
+     "hybrid": [
+         {}, // First Query
+         {}, // Second Query
+         ... // Other Queries
+     ] 
+   }
+}
+```
+
+## Benchmarking z-score normalization performance
+
+Benchmark experiments were conducted on an OpenSearch cluster with one **r6g.8xlarge** instance as the coordinator node and three **r6g.8xlarge** instances as data nodes. To evaluate z-score normalization comprehensively, we measured two key metrics across five distinct datasets.  
+
+For more information about the datasets used, see [Datasets](https://public.ukp.informatik.tu-darmstadt.de/thakur/BEIR/datasets/).
+
+### Sample queries and passages
+
+The following table provides sample queries and passages for each dataset.
+
+|Dataset	|Sample query	|Sample passage	|
+|:---	|:---	|:---	|
+|SciDocs	|CFD Analysis of Convective Heat Transfer Coefficient on External Surfaces of Buildings	|`This paper provides an overview of the application of CFD in building performance simulation for the outdoor environment, focused on four topics...`	|
+|FiQA	|“Business day” and “due date” for bills	|`I don't believe Saturday is a business day either. When I deposit a check at a bank's drive-in after 4pm Friday, the receipt tells me it will credit as if I deposited on Monday. If a business' computer doesn't adjust their billing to have a weekday due date ...	`|
```

Comment:
```suggestion
|FiQA	|"Business day" and "due date" for bills	|`I don't believe Saturday is a business day either. When I deposit a check at a bank's drive-in after 4pm Friday, the receipt tells me it will credit as if I deposited on Monday. If a business' computer doesn't adjust their billing to have a weekday due date ...	`|
```